### ISSUE TITLE
Compiler doesn't show error when code is trying to assign something to getter/setter

### ISSUE THREAD

**Issue by bobrosoft** [Microsoft/TypeScript#9928]
**TypeScript Version:**  1.8.0

**Code**

```ts
class Something {
	get x(): string {
		return this._x + " from getter";
	}
	
	constructor(private _x: string) {
		
	}
	
	doSomething(newX: string) {
		this.x = newX; // here should be compilation error
	}
}

let s = new Something("foo");
s.doSomething("bar");
alert(s.x);
```

**Expected behavior:**
Should throw compilation error on marked line.

**Actual behavior:** 
Throws no errors or something, but code will not work in a way programmer intended it to.

**Additional notes:**
I found that issue when I was trying to refactor some code to use private setter with public getter (pretty common case, I think). During refactoring that could lead to multiple silent issues. Apple Swift has made it very simple to use as `public private(set) var prop: String` (private setters, very handy), but in TS I found only way above to do something like this.

**Comment by RyanCavanaugh** [comment]
Please read the FAQ before posting new issue. https://github.com/Microsoft/TypeScript/wiki/FAQ#why-are-getters-without-setters-not-considered-read-only

This has been addressed with #12 and you'll now see the error
>  error TS2450: Left-hand side of assignment expression cannot be a constant or a read-only property.

**Comment by bobrosoft** [comment]
@RyanCavanaugh oh, sorry, didn't find it. But, can you summarize on it as I didn't get on final solution for that: is it going to be released in TypeScript 2.0 or that's not the issue (i.e. "by design")?

**Comment by RyanCavanaugh** [comment]
In TS 2.0 you'll see the error I showed above.
